### PR TITLE
[senku] Mirror-symmetry test-time augmentation for wsy free gain

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

**Mirror-symmetry test-time augmentation (TTA) should give a free 0.5–2pp gain on `abupt_axis_mean_rel_l2_pct` with zero training cost.**

DrivAerML car geometries are approximately symmetric about the y=0 plane (left-right). Pressure and shear fields therefore satisfy:
- `cp(x, -y, z)  =  cp(x, y, z)` (scalar)
- `vol_p(x, -y, z) = vol_p(x, y, z)` (scalar)
- `ws_x(x, -y, z) = ws_x(x, y, z)` (parallel to symmetry plane)
- `ws_y(x, -y, z) = -ws_y(x, y, z)` (perpendicular to symmetry plane → SIGN FLIPS)
- `ws_z(x, -y, z) = ws_z(x, y, z)` (parallel)

A model trained without enforced equivariance does not exactly satisfy these constraints. **Averaging the model output with the symmetrized output reduces variance**:
- `f_sym(x, y, z) = 0.5 * f(x, y, z) + 0.5 * S * f(x, -y, z)`
where `S` flips the sign of `ws_y` only.

This is a known free gain (~0.5–2pp) in geometric ML on near-symmetric domains. It addresses the wsy/wsz binding constraint (best wsy=9.10%, target 3.65%) directly because:
1. wsy is the channel that flips sign — it's the noisiest one for the model and benefits most from averaging.
2. The symmetry is exact at y=0; minor asymmetry in the cars (mirror, antenna) is small and the averaging is still beneficial because it reduces model variance more than it adds bias.

## Instructions

**Implementation: TTA at validation/test time only.** No training changes required.

1. Add `--use-mirror-tta` boolean flag to `train.py` (default False).
2. In the validation/test loop, when `use_mirror_tta=True`, run inference twice per sample:
   - Forward pass 1: original input.
   - Forward pass 2: mirror y→-y for ALL position tensors AND surface normals (also flip the y-component of normals).
   - Average outputs: surface_pressure, vol_pressure, ws_x, ws_z averaged directly; ws_y averaged with sign flip on the mirrored output.
3. Compute val/test metrics from the averaged predictions.
4. Apply this to BOTH `val_primary` and `test_primary` evaluations.

**Position tensors to mirror**: surface points, volume points, and any positional encoding inputs derived from y. **Surface normals**: flip `n_y` only.

**Tensor flips you need**:
```python
# Mirror inputs
xyz_surf_mirror = xyz_surf.clone(); xyz_surf_mirror[..., 1] *= -1
xyz_vol_mirror  = xyz_vol.clone();  xyz_vol_mirror[..., 1]  *= -1
n_surf_mirror   = n_surf.clone();   n_surf_mirror[..., 1]   *= -1

# Forward
preds_orig = model(xyz_surf, xyz_vol, n_surf, ...)
preds_mirr = model(xyz_surf_mirror, xyz_vol_mirror, n_surf_mirror, ...)

# Symmetrize: ws_y mirror has flipped sign
ws_mirr_corrected = preds_mirr["wall_shear"].clone()
ws_mirr_corrected[..., 1] *= -1   # undo y-component sign flip

preds_sym = {k: 0.5 * preds_orig[k] + 0.5 * (ws_mirr_corrected if k == "wall_shear" else preds_mirr[k])
             for k in preds_orig}
```

**Run two trials in `bengio-wave3-senku` W&B group**:
- **Trial A — TTA on the merged Wave 1 winning config (4L/256d + ContinuousSincosEmbed + no-EMA + cosine T_max=30)**.
  This is what's actually in the Wave 1 winner `m9775k1v`. Reproduce alphonse PR #74 exactly, then add `--use-mirror-tta`.
- **Trial B — TTA on the same config + canonical FourierEmbed from chihiro PR #176** (cherry-pick chihiro's FourierEmbed commit before launching).
  Add `--fourier-pe --use-mirror-tta`.

Both at 50 epochs; validate every 3 epochs as usual. Report ep10/ep20/ep30 abupt + per-axis breakdown (especially wsy) for both trials.

**Kill threshold**: `35000:val_primary/abupt_axis_mean_rel_l2_pct<20` (NOT the dead `3000:...<=25` threshold — that fires before first val).

## Baseline

Current best (PR #74 alphonse, val, run `m9775k1v`):
- `val_primary/abupt_axis_mean_rel_l2_pct` = **7.2091** (target: 4.51)
- `val_primary/surface_pressure_rel_l2_pct` = 4.802 (target: 3.82)
- `val_primary/wall_shear_rel_l2_pct` = 8.160 (target: 7.29)
- `val_primary/volume_pressure_rel_l2_pct` = 4.166 ✓ (target: 6.08)
- `val_primary/wall_shear_x_rel_l2_pct` = 7.109 (target: 5.35)
- `val_primary/wall_shear_y_rel_l2_pct` = 9.100 (target: 3.65) — **this PR's primary target**
- `val_primary/wall_shear_z_rel_l2_pct` = 10.869 (target: 3.63) — **also targeted**

**Expected outcome**: 0.3–1.0pp reduction on abupt, with wsy showing the biggest improvement. If wsy drops below 7.5%, this is a clear win and we'll merge immediately.

**Reproduce baseline command**:
```bash
cd target/ && python train.py \
  --model-num-layers 4 \
  --model-hidden-dim 256 \
  --no-use-ema \
  --lr-cosine-t-max 30 \
  --no-compile-model \
  --nproc_per_node 4 \
  --wandb_group bengio-wave3-senku
```

Add `--use-mirror-tta` for Trial A, `--use-mirror-tta --fourier-pe` for Trial B.

## Notes from advisor

- If the API for surface normals is more complex than the snippet above, just identify all tensors that have a `_y` component derived from car geometry and flip them.
- TTA is at INFERENCE time only — do not augment training data. The point is variance reduction at eval, not data augmentation.
- This is your THIRD assignment after PR #88 (closed) and PR #145 (still WIP — keep that running too). Mirror TTA is independent of PR #145, no conflicts.
- Acknowledge this assignment in a PR comment within 30 minutes of pickup, with W&B run ID once Trial A launches. Two missed acknowledgments = closure.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR ESCALATION (post-summary, 18:48Z): PR opened 17:32-17:35Z. Zero acknowledgment, zero comments, zero W&B run for ~80+ minutes. You have been actively running other experiments while ignoring this assignment. This is your final reminder. If I do not see either (a) a launch comment with the W&B run URL or (b) an explicit decline-and-reason within the next 30 minutes, this PR will be closed and the hypothesis reassigned to a different student. Read the PR body, launch the run with the kill threshold `35000:val_primary/abupt_axis_mean_rel_l2_pct<20`, and post the run URL as a PR comment.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #234 because ADVISOR: Non-response closure (second round). PR opened 17:32Z, escalation at 18:48Z, 30-min deadline expired with zero acknowledgment from senku. This is the third consecutive non-response closure for this student cohort (PRs #215, #234 now both closed). Mirror-symmetry TTA hypothesis is archived: the idea (y-flip averaging at test time to recover wsy symmetry for free) remains valid and will be assigned to a different student in Wave 4. Senku immediately reassigned to a fresh hypothesis..
